subprojects/core/src/main/java/org/gradle/api/internal/changedetection/state/mirror/logical/DefaultFileCollectionFingerprint.java

```diff
@@ -38,20 +38,22 @@
 
 public class DefaultFileCollectionFingerprint implements FileCollectionSnapshot {
 
+    private final FingerprintingStrategy.Identifier fingerprintingStrategyIdentifier;
     private final FingerprintCompareStrategy strategy;
     private final Map<String, NormalizedFileSnapshot> snapshots;
     private final Iterable<PhysicalSnapshot> roots;
     private HashCode hash;
 
     public DefaultFileCollectionFingerprint(FingerprintCompareStrategy strategy, Map<String, NormalizedFileSnapshot> snapshots, @Nullable HashCode hash) {
-        this(strategy, snapshots, hash, null);
+        this(null, strategy, snapshots, hash, null);
     }
 
     public DefaultFileCollectionFingerprint(FingerprintingStrategy strategy, Iterable<PhysicalSnapshot> roots) {
-        this(strategy.getCompareStrategy(), strategy.collectSnapshots(roots), null, roots);
+        this(strategy.getIdentifier(), strategy.getCompareStrategy(), strategy.collectSnapshots(roots), null, roots);
```

Comment:
`CompareStrategy` is now an enum. Would that be enough for the scan plugin? Then it wouldn't be possible to distinguish between compile and runtime classpath.

Reply:
No, we need the extra detail.